sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -775,4 +783,120 @@ export class ApplicationControllerService {
       throw error;
     }
   }
+
+  /**
+   * Validate the application submission and get the validated result.
+   * @param applicationId application ID.
+   * @param studentId student used for authorization.
+   * @param payload payload to create the application.
+   * @throws UnprocessableEntityException if the program year is not active.
+   * @throws BadRequestException if the application data is invalid, failing the druRun submission.
+   */
+  async validateApplicationSubmission(
+    applicationId: number,
+    studentId: number,
+    payload: SaveApplicationAPIInDTO,
+  ): Promise<ApplicationData> {
+    const application =
+      await this.applicationService.getApplicationForSubmissionValidation(
+        applicationId,
+        studentId,
+      );
+    if (!application) {
+      throw new NotFoundException("Application not found.");
+    }
+    if (!application.programYear.active) {
+      throw new UnprocessableEntityException("Program year is not active.");
+    }
+    // Validate the values in the submitted application before submitting.
+    await this.validateSubmitApplicationData(payload);
+    const submissionResult =
+      await this.formService.dryRunSubmission<ApplicationData>(
+        application.programYear.formName,
+        payload.data,
+      );
+    if (!submissionResult.valid) {
+      throw new BadRequestException(
+        "Not able to create an application due to an invalid request.",
+      );
+    }
+    await this.offeringIntensityRestrictionCheck(
+      studentId,
+      submissionResult.data.data.howWillYouBeAttendingTheProgram,
```

Comment:
Is expected to come from `application.offeringIntensity` from @sh16011993 PR ?

Reply:
Adjusted now.